transforms/transforms_test.go

```diff
@@ -362,6 +363,45 @@ func TestUniqueErrorForInvalidHeader(t *testing.T) {
 	}
 }
 
+func TestGroupBy(t *testing.T) {
+	transform := GroupBy(KeyIdentifier("group"))
+	input := []optimus.Row{
+		{"group": "1", "key": "a"},
+		{"group": "2", "key": "d"},
+		{"group": "3", "key": "g"},
+		{"group": "1", "key": 1},
+		{"group": "2", "key": 2},
+		{"group": "3", "key": 3},
+		{"group": "1", "key": 'b'},
+		{"group": "2", "key": 'e'},
+		{"group": "3", "key": 'h'},
+	}
+	expected := []optimus.Row{
+		{"1": []optimus.Row{input[0], input[3], input[6]}},
```

Comment:
Yeah, I think it makes sense when it's one dict, but I don't think we want to output it that way - I think we want to output each group as a separate object, and then I don't think that interface makes sense.

I have a slight bias against `value` because that seems vague to me - if I was told that there was a `value` key on the `GroupBy` object and not what the other keys were, I wouldn't know if it was the rows or the identifier for the group.

I think my preferred output is `{"id": "1", "rows": []optimus.Row{...}}`, but I think any of these options are good. Pick whatever you like, @jiancheung :)